app/models/grda_warehouse/monitoring/metric_snapshot.rb

```diff
@@ -0,0 +1,64 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module GrdaWarehouse::Monitoring
+  class MetricSnapshot < GrdaWarehouseBase
+    self.table_name = 'metric_snapshots'
+
+    belongs_to :entity, polymorphic: true
+    belongs_to :metric_definition,
+               class_name: 'GrdaWarehouse::Monitoring::MetricDefinition'
+
+    validates :initial_observation_date, presence: true
+    validates :current_observation_date, presence: true
+    validates :initial_value, presence: true
+    validates :current_value, presence: true
+
+    validates :current_observation_date, uniqueness: {
+      scope: [:entity_type, :entity_id, :metric_definition_id],
+      message: 'Only one active snapshot per entity/metric',
+    }
```

Comment:
Users never actually enter snapshots right? We are doing bulk inserts. So these just mirror db constraints but are unused(?). Suggest just removing them in that case